### PR TITLE
refactor(asr): rename `Character` ASR node to `String`

### PR DESCRIPTION
**Pull request by kmr-srbh**

### CODE REVIEW

File: tests/reference/wat-wasm_unary_minus-81a29ff.stderr
Comment (kmr-srbh) [inline tests/reference/wat-wasm_unary_minus-81a29ff.stderr:1]: @gxyd could you please confirm if you are aware of these error references being generated with the Simplifier pass branch?

Comment (gxyd) [reply tests/reference/wat-wasm_unary_minus-81a29ff.stderr:1]: I noticed the same a few days back, when checking the status of tests presents in `tests.toml`. Ofcourse, these errors aren't ideal, but we'll definitely need a closer look for the real issue behind these.

Just like we are working on *integration tests* currently, we'll need to look at the tests in `tests.toml` as well.

Comment (kmr-srbh) [reply tests/reference/wat-wasm_unary_minus-81a29ff.stderr:1]: Thanks for confirming this! :)

Comment (certik) [reply tests/reference/wat-wasm_unary_minus-81a29ff.stderr:1]: Yes, integration tests are more important. The reference tests will add a lot more file noise, so I would focus on them at the very end.

**Comment by certik** [comment]
Can you remove the reference tests changes? Otherwise it's going to make the main simplifier PR a lot more complicated. We'll focus on reference tests later.

**Comment by kmr-srbh** [comment]
Yes, I will remove them today. :+1:

**Comment by kmr-srbh** [comment]
Closing this PR to clear the reference files. They are fairly large in number. I will verify the references and open a new PR as we do need some new reference for the `Character` to `String` change.

**Pull request closed without merging**